catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/ResourceCache.java

```diff
@@ -321,13 +325,13 @@ public Resource getValid(String key, Metacard latestMetacard) {
                         "Entry found in the cache, but no product found in cache directory for key = {}",
                         key);
                 cache.remove(key);
-                throw new IllegalArgumentException(
+                throw new CacheException(
                         "Entry found in the cache, but no product found in cache directory for key = "
                                 + key);
             }
         } else {
             LOGGER.debug("No product found in cache for key = {}", key);
-            throw new IllegalArgumentException("No product found in cache for key = " + key);
+            throw new CacheException("No product found in cache for key = " + key);
         }
```

Comment:
This is a cache miss, which isn't an exceptional situation, so I would expect us to simply return null.  It smells like we're using Exceptions for normal program flow.